libopenage/gamestate/game_main.cpp

```diff
@@ -37,17 +38,14 @@ GameMain::GameMain(const Generator &generator)
 }
 
 GameMain::~GameMain() {
-	for (auto &p : this->players) {
-		delete p;
-	}
 }
 
 unsigned int GameMain::player_count() const {
 	return this->players.size();
 }
 
 Player *GameMain::get_player(unsigned int player_id) {
-	return this->players.at(player_id);
+	return this->players.at(player_id).get();
```

Comment:
What i have tried was to change the return type to ```std::shared_ptr<Player>``` - it did not fail. 
That means, it seems not to be called, or using auto all over the place. 

So my question is, if it were to break any existing code to change it